homeassistant/components/recorder/__init__.py

```diff
@@ -488,15 +486,13 @@ def _setup_connection(self):
         """Ensure database is ready to fly."""
         kwargs = {}
 
-        # pylint: disable=unused-variable
-        @listens_for(Engine, "connect")
-        def setup_connection(dbapi_connection, connection_record):
+        def setup_recorder_connection(dbapi_connection, connection_record):
             """Dbapi specific connection settings."""
 
             # We do not import sqlite3 here so mysql/other
             # users do not have to pay for it to be loaded in
             # memory
-            if self.db_url == "sqlite://" or ":memory:" in self.db_url:
```

Comment:
This changes behavior. 

If you specify just `sqlite://`, you get a SQLite in-memory database. If it starts with `sqlite://`, it can also be file based, like the default.

During tests we run with an in-memory DB.

Reply:
That logic should still be below.  This fixes a regression I caused in b09b5729a36e69f517703038a38e197ff5c13f42